packages/ui/Scroller/Scroller.js

```diff
@@ -27,128 +27,95 @@ import css from './Scroller.module.less';
  * @ui
  * @public
  */
-class ScrollerBase extends Component {
-	static displayName = 'ui:ScrollerBase'
-
-	static propTypes = /** @lends ui/Scroller.ScrollerBase.prototype */ {
-		children: PropTypes.node.isRequired,
-
-		/**
-		 * Callback method of scrollTo.
-		 * Normally, `Scrollable` should set this value.
-		 *
-		 * @type {Function}
-		 * @private
-		 */
-		cbScrollTo: PropTypes.func,
-
-		/**
-		 * Direction of the scroller.
-		 *
-		 * Valid values are:
-		 * * `'both'`,
-		 * * `'horizontal'`, and
-		 * * `'vertical'`.
-		 *
-		 * @type {String}
-		 * @default 'both'
-		 * @public
-		 */
-		direction: PropTypes.oneOf(['both', 'horizontal', 'vertical']),
-
-		/**
-		 * Prop to check context value if Scrollbar exists or not.
-		 *
-		 * @type {Boolean}
-		 * @private
-		 */
-		isVerticalScrollbarVisible: PropTypes.bool,
-
-		/**
-		 * `true` if RTL, `false` if LTR.
-		 *
-		 * @type {Boolean}
-		 * @private
-		 */
-		rtl: PropTypes.bool
-	}
-
-	static defaultProps = {
-		direction: 'both'
-	}
-
-	constructor (props) {
-		super(props);
-
-		this.containerRef = React.createRef();
-	}
-
-	componentDidMount () {
-		this.calculateMetrics();
-	}
-
-	componentDidUpdate (prevProps) {
-		this.calculateMetrics();
-		if (this.props.isVerticalScrollbarVisible && !prevProps.isVerticalScrollbarVisible) {
-			this.forceUpdate();
+let ScrollerBase = (props, reference) => {
+	// constructor (props) {
+	const containerRef = useRef();
+	const [, forceUpdate] = useReducer(x => x + 1, 0);
+
+	useEffect(() => {
+		// componentDidMount
+		calculateMetrics();
+	}, []);
+
+	useEffect(() => {
+		// componentDidUpdate
+		forceUpdate();
```

Comment:
This code introduced from #1618 and there is a possibility to cover the original issue in another way now. Until we have an alternative solution, let's keep this for now.

Reply:
I just added description here. I'll check later this code is needed still or not.